src/Core/Http/Input.php

```diff
@@ -41,51 +42,46 @@ private static function put(): array
     }
 
     /**
+     * extract data from submit form as form data
      * @param $file_input
-     * @return array|string[]
+     * @return array
      */
-    private static function extractFromFormData($file_input)
+    private static function extractFromFormData($file_input): array
     {
-        $fragma = [];
+        $form_data_fragmentation = [];
         $explode = explode("\r", implode("\r", explode("\n", $file_input)));
         $len_Arr = count($explode);
         for ($i = 1; $i < $len_Arr; $i++) {
             if (!strchr($explode[$i], "----------------------------")) {
                 if (strlen($explode[$i]) > 0) {
                     $replaced = str_replace("Content-Disposition: form-data; name=", "", $explode[$i]);
-                    array_push($fragma, $replaced);
+                    $form_data_fragmentation[] = $replaced;
                 }
             }
         }
-        $len_object = count($fragma);
+        $len_object = count($form_data_fragmentation);
```

Comment:
Load data from form-data should be revised, refactoring and validate while load form-data submission.
The process is not clean and testable.

Reply:
you're right